diesel/src/pg/connection/raw.rs

```diff
@@ -183,6 +183,41 @@ impl RawConnection {
             ))
         }
     }
+
+    pub(super) fn pqnotifies(&self) -> Option<PGNotification> {
+        let conn = self.internal_connection;
+        let _ = unsafe { PQconsumeInput(conn.as_ptr()) };
```

Comment:
we should check the returned integer here to check if an error occurred. That likely means that the notification iterator returns a `QueryResult<PgNotification>` instead of just a `PgNotification>`.

It should make it also easier to handle the possible UTF-8 errors below.

Reply:
I'm not sure how I could nicely convert the UTF-8 errors into DatabaseError.

Reply:
I would rather use the [`DeserializationError`](https://docs.diesel.rs/2.2.x/diesel/result/enum.Error.html#variant.DeserializationError) variant for that.